--- a/third_party/phpmailer/PHPMailerAutoload.php
+++ b/third_party/phpmailer/PHPMailerAutoload.php
@@ -30,20 +30,23 @@ function PHPMailerAutoload($classname)
     }
 }
 
-if (version_compare(PHP_VERSION, '5.1.2', '>=')) {
-    //SPL autoloading was introduced in PHP 5.1.2
-    if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
-        spl_autoload_register('PHPMailerAutoload', true, true);
-    } else {
-        spl_autoload_register('PHPMailerAutoload');
-    }
+if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
+    spl_autoload_register('PHPMailerAutoload', true, true);
 } else {
-    /**
-     * Fall back to traditional autoload for old PHP versions
-     * @param string $classname The name of the class to load
-     */
-    function __autoload($classname)
-    {
-        PHPMailerAutoload($classname);
-    }
+    spl_autoload_register('PHPMailerAutoload');
 }
+
+/**
+ * if (version_compare(PHP_VERSION, '5.1.2', '>=')) {
+ *    //SPL autoloading was introduced in PHP 5.1.2
+ *    if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
+ *        spl_autoload_register('PHPMailerAutoload', true, true);
+ *    } else {
+ *        spl_autoload_register('PHPMailerAutoload');
+ *    }
+ *  else {
+ *    function __autoload($classname)
+ *    {
+ *        PHPMailerAutoload($classname);
+ *    }
+ */
