--- a/lib/api/rest/v1/tlRestApi.class.php
+++ b/lib/api/rest/v1/tlRestApi.class.php
@@ -261,7 +261,7 @@ class tlRestApi
 
     if( !is_null($tproject) )
     {
-      $items = $this->tprojectMgr->get_all_testplans($tproject[0]['id']);
+      $items = $this->tprojectMgr->get_all_testplans($tproject['id']);
       $op['items'] = (!is_null($items) && count($items) > 0) ? $items : null;
     }
     else 
@@ -285,11 +285,11 @@ class tlRestApi
   {
     $op  = array('status' => 'ok', 'message' => 'ok', 'items' => null);
     $tproject = $this->getProjects($idCard, array('output' => 'internal'));
-
+    
     if( !is_null($tproject) )
     {
       $tcaseIDSet = array();
-      $this->tprojectMgr->get_all_testcases_id($tproject[0]['id'],$tcaseIDSet);
+      $this->tprojectMgr->get_all_testcases_id($tproject['id'],$tcaseIDSet);
       if( !is_null($tcaseIDSet) && count($tcaseIDSet) > 0 )
       {
         $op['items'] = array();
@@ -298,7 +298,7 @@ class tlRestApi
           $item = $this->tcaseMgr->get_last_version_info($tcaseID);
           $item['keywords'] = $this->tcaseMgr->get_keywords_map($tcaseID);
           $item['customfields'] = $this->tcaseMgr->get_linked_cfields_at_design($tcaseID,$item['tcversion_id'],
-                                                                                null,null,$tproject[0]['id']);
+                                                                                null,null,$tproject['id']);
           $op['items'][] = $item;
         }
       }
